Makefile

```diff
@@ -95,14 +95,17 @@ $(VMLINUXH): $(OUTPUT)
 		echo "ERROR: kernel does not seem to support BTF"; \
 		exit 1; \
 	fi
-	@if [ ! -f $(VMLINUXH) ]; then \
-		if [ ! $(BPFTOOL) ]; then \
-			echo "ERROR: could not find bpftool"; \
-			exit 1; \
-		fi; \
-		echo "INFO: generating $(VMLINUXH) from $(BTFFILE)"; \
-		$(BPFTOOL) btf dump file $(BTFFILE) format c > $(VMLINUXH); \
-	fi
+	@if [ ! $(BPFTOOL) ]; then \
+		echo "ERROR: could not find bpftool"; \
+		exit 1; \
+	fi;
```

Comment:
Despite what was suggested in https://github.com/aquasecurity/libbpfgo/pull/239#discussion_r966337865, this remains for now. Whether we keep an up-to-date vmlinux.h in the tree or build (and make available for consumer) the latest bpftool can be discussed in #177.

My thoughts anyway: the consumer of this library needs, for better experience, an updated tooling (bpftool in this case), which is not offered in common distros, but those that are rolling release.

Reply:
Question: why to "fix" this instead of simply adding a vmlinux.h file to the repository (from a recent enough kernel, like 5.15 or so. We can have the file "tar.bz2" and uncompressed by Makefile to save space even. @grantseltzer WDYT ? Obligating the user to have bpftool is complicated. If we pick Ubuntu distribution, for example, it depends on having package named "linux-tools-XXXX" (depending on the kernel version), and is error-prone (sometimes bpftool in a container has wrong version due to host kernel being different, and things like that).

Reply:
Agreeing with you both. I think it demands its own issue, but first let's decides the requirements.

- Remove libbpfgo Makefile bpftool requirement.
- Make available an updated pre-made vmlinux.h (I think it can be one extracted from Arch/Manjaro - due to its freshness) in the libbpfgo tree for static and dynamic compilations (format to be defined).
- Add a Makefile flag (env var) to one specify its own vmlinux.h overriding the offered by libbpfgo.

WDYT?

Reply:
@geyslan That sounds great to me!

Reply:
As https://github.com/aquasecurity/libbpfgo/issues/242 was created, I think this PR is ready for review.

Reply:
Done: https://github.com/aquasecurity/libbpfgo/issues/242.